lib/todo_or_die.rb

```diff
@@ -38,14 +52,22 @@ def self.__clean_backtrace(stack)
 end
 
 # The main event
-def TodoOrDie(message, by: by_omitted = true, if: if_omitted = true) # rubocop:disable Naming/MethodName
+def TodoOrDie(message, by: by_omitted = true, if: if_omitted = true, warn_by: warn_by_omitted = true) # rubocop:disable Naming/MethodName
   due_at = Time.parse(by.to_s) unless by_omitted
   is_due = by_omitted || Time.now > due_at
   condition = binding.local_variable_get(:if) unless if_omitted
   condition_met = if_omitted || (condition.respond_to?(:call) ? condition.call : condition)
 
   if is_due && condition_met
     die = TodoOrDie.config[:die]
-    die.call(*[message, due_at, condition].take(die.arity.abs))
+    return die.call(*[message, due_at, condition].take(die.arity.abs))
+  end
```

Comment:
In order to avoid multiple return sites in this method, would you consider an `elsif is_warn` here that pulled up any necessary assignment above the `if` branch?

Also, `is_warn` is not as meaningful a name as `is_due`; maybe change both? `should_die` and `should_warn`?